### PR TITLE
Add data-test attribute to verify contact skip button.

### LINKED ISSUE #7836: [Regression] v1 React AmplifyVerifyContact component does not include data-test attribute on Skip button
**Describe the bug**
`data-test` attribute was added to amplify-verify-contact a couple years ago but is missing with the newer component.
See: https://github.com/aws-amplify/amplify-js/pull/3253/files#diff-33977785c6d5a2d616cc9a4065e13d4a976f6864e045f140aa5148ef4952c714R155

Now it's missing: https://github.com/aws-amplify/amplify-js/blob/6d1bed8ec59ae4a44f296b90c2b9aa781c5f68b3/packages/amplify-ui-components/src/components/amplify-verify-contact/amplify-verify-contact.tsx#L159

**To Reproduce**

1. Create Cognito user with un-verified email
2. Create Amplify UI React authentication component
3. Run app and enter test user credentials
4. View page source or verification confirmation modal element source to see there is no data-test attribute

**Expected behavior**
data-test attribute is expected for Skip anchor just like all the other authentication elements.

The branching around the new components is not something I can follow accurately to attempt to contribute a fix. `main` branch appears to be the old structure where the Skip element does include the data-test attribute.
https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Auth/VerifyContact.tsx#L189
[current point-in-time] https://github.com/aws-amplify/amplify-js/blob/be53c6187057728a9e2cf24265324fd1f470e5e0/packages/aws-amplify-react/src/Auth/VerifyContact.tsx#L189

### PR DESCRIPTION
**Pull request by reid-harrison**
Fix unique data-test attribute on verify contact phone number radio.

Closes #7836

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by wlee221 — COMMENTED** [review]
Hi @reid-harrison! The change itself looks good to me, but we did want to point out that `data-test` attributes are for internal integration testing. May I ask what your use case is so that we are better aware of what we want to support?

**Review by ericclemmons — APPROVED** [review]
Thanks for the contribution!

I assume you're relying on these for integration tests, like with Cypress?

If so, I'm in favor of merging this in.

**Comment by reid-harrison** [comment]
Yes, my situation is kind of simple but I was primarily just following the [Cypress tutorial for Amplify](https://aws.amazon.com/blogs/mobile/run-end-to-end-cypress-tests-for-your-fullstack-ci-cd-deployment-with-amplify-console/) and using a test account with a fake, unverified email. Even though I have verification disabled in the User Pool, Amplify Authenticator still prompts with the verification component so I went down a rabbit hole just looking to click the skip link. In the mean time, I used the AWS CLI to just verify the test user: `aws cognito-idp admin-update-user-attributes --username test --user-attributes Name=email_verified,Value=true --user-pool-id #{user_pool_id}`

cognito_verification_disabled

I wasn't sure if it would be preferred to name the data-test attribute for the skip element `verify-contact-skip-button` or `verify-contact-skip-link` and I've since seen it appears `*-link` is the preference for anchor variants so I'd be happy to change it if desired.

File: packages/amplify-ui-components/src/components/amplify-verify-contact/amplify-verify-contact.tsx [hunk 1]
```diff
@@ -139,7 +139,7 @@ export class AmplifyVerifyContact {
             value="phone_number"
             handleInputChange={event => this.handleInputChange(event)}
             inputProps={{
-              'data-test': 'verify-contact-email-radio',
+              'data-test': 'verify-contact-phone-number-radio',
```
Comment (wlee221) [inline packages/amplify-ui-components/src/components/amplify-verify-contact/amplify-verify-contact.tsx:142]: Nice catch!

**Review by wlee221 — COMMENTED** [review]
Yes, I agree that we should go with `verify-contact-skip-link`. Otherwise, I'm in favor of merging this in!

File: packages/amplify-ui-components/src/components/amplify-verify-contact/amplify-verify-contact.tsx [hunk 2]
```diff
@@ -159,6 +159,7 @@ export class AmplifyVerifyContact {
               <amplify-button
                 variant="anchor"
                 onClick={() => this.handleAuthStateChange(AuthState.SignedIn, this.user)}
+                data-test="verify-contact-skip-button"
```
Comment (wlee221) [inline packages/amplify-ui-components/src/components/amplify-verify-contact/amplify-verify-contact.tsx:162]:
```suggestion
                data-test="verify-contact-skip-link"
```

**Review by wlee221 — APPROVED** [review]
Let's just get this changed, and we'll get this merged in!

**Pull request merged**